config/routes.rb

```diff
@@ -1,8 +1,28 @@
 Rails.application.routes.draw do
+  devise_for :users
+
+  devise_scope :user do
+    get '/users/sign_out' => 'devise/sessions#destroy'
+  end
 
   root "books#index"
 
-  #get "/books/:id", to: "books#show"
   resources :books
+  resources :authors
+
+  resources :users do
+    resources :lists
+  end
+
+  get "/search", to: "books#search"
+
+  get "/favorites/:id", to: "favorites#destroy"
+  get "/favorites", to: "favorites#create"
```

Comment:
absolutely not.
avoid this "get", "post" routes at all cost
`resources :favorites, only: %i[index create destroy]` would do the same, but more elegantly

Reply:
same lower